src/services/base.service.js

```diff
@@ -51,6 +108,73 @@ const baseSearchById = async (args, resourceType) => {
   return new dataType(result);
 };
 
+/**
+ * Searches using query parameters.
+ * @param {*} args The args from the request.
+ * @param {*} req The Express request object. This is used by the query builder.
+ * @param {*} resourceType The resource type we are searching on.
+ * @param {*} paramDefs Optional parameter definitions for the specific resource types. Specific
+ *                      resource services should call this and pass
+ * @returns Search set result bundle
+ */
+const baseSearch = async (args, { req }, resourceType, paramDefs = {}) => {
+  // grab the schemas for the data type and Bundle to use for response
+  const dataType = resolveSchema(args.base_version, resourceType.toLowerCase());
+  const Bundle = resolveSchema(args.base_version, 'bundle');
+
+  // wipe out params since the 'base_version' here breaks the query building
+  req.params = {};
+
+  // build result bundle. default to an empty result
+  const searchBundle = new Bundle({
+    type: 'searchset',
+    meta: { lastUpdated: new Date().toISOString() },
+    total: 0
+  });
+
+  // build the aggregation query
+  const filter = qb.buildSearchQuery({ req: req, includeArchived: true, parameterDefinitions: paramDefs });
+
+  // if the query builder was able to build a query actually execute it.
+  if (filter.query) {
+    // grab the results from aggregation. has metadata about counts and data with resources
+    const results = (await (await findResourcesWithAggregation(filter.query, resourceType)).toArray())[0];
+
+    // If this is undefined, there are no results.
+    if (results.metadata[0]) {
```

Comment:
is there a case where `results` could be undefined? should we add a `results && results.metadata[0]` case instead?

Possibly not, just spitballin lol

Reply:
I had a similar question but wasn't sure if it was just a really rare edge case.

Reply:
Good call. It is possible, but seems to only happen when the query building comes up with a goose egg which it also returns an error.